### PR TITLE
[SofaConstraint] ADD Data to show constraint forces

### PR DESCRIPTION
**Pull request by zhongkaidefrost**
from comment:
`The Data constraintForces is used to expose the intensities of constraint forces in the simulation. We use the constraint forces as the approximation of real contact forces. We use this Data to provide contact information for the catheter insertion. The user can easily check the constraint forces from the GenericConstraint component interface.
`

______________________________________________________

This PR: 
- [x] builds with SUCCESS for all platforms on the CI.
- [x] does not generate new warnings.
- [x] does not generate new unit test failures.
- [x] does not generate new scene test failures.
- [x] does not break API compatibility.
- [x] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

File: modules/SofaConstraint/GenericConstraintSolver.h [hunk 1]
```diff
@@ -130,6 +130,8 @@ class SOFA_CONSTRAINT_API GenericConstraintSolver : public ConstraintSolverImpl
 	Data<int> currentIterations; ///< OUTPUT: current number of constraint groups
 	Data<double> currentError; ///< OUTPUT: current error
     Data<bool> reverseAccumulateOrder; ///< True to accumulate constraints from nodes in reversed order (can be necessary when using multi-mappings or interaction constraints not following the node hierarchy)
+    Data<helper::vector< double >> constraintForces;
```
Comment (guparan) [inline modules/SofaConstraint/GenericConstraintSolver.h]: Please use `d_` convention when defining new Data.
See [N5] in https://github.com/sofa-framework/sofa/blob/master/GUIDELINES.md#naming

**Review by guparan — CHANGES_REQUESTED** [review]

**Comment by guparan** [comment]
Congrats for your first PR @zhongkaidefrost ! :clap: 
I just suggested a change, please check it when you have a few minutes :-)

**Review by guparan — APPROVED** [review]

**Comment by epernod** [comment]
@zhongkaidefrost thanks for your PR.
I'm not sure to understand what the Data constraintForces is used for as it is filled but never used. 
Could you explain in the description of your PR why you introduced this new Data.

In the same idea, it would be nice if you could add an inline doxygen comment to your Data.

**Comment by zhongkaidefrost** [comment]
The Data constraintForces is used to provide the intensities of constraint forces in the simulation. We use the constraint forces as the approximation of real contact forces. We use this Data to provide contact information for the catheter insertion. The user can easily check the constraint forces from the GenericConstraint component interface.

**Review by epernod — APPROVED** [review]

**Review by epernod — APPROVED** [review]

**Comment by damienmarchal** [comment]
[ci-build] (because of weird CI-failure)

**Comment by epernod** [comment]
[ci-build][with-all-tests]

**Comment by guparan** [comment]
@damienmarchal Do you still plan to change the variable name?

**Comment by damienmarchal** [comment]
I forgot :)

EDIT: Done

**Comment by damienmarchal** [comment]
@guparan ready/merge ?

**Pull request merged**